group-coordinator/src/test/java/org/apache/kafka/coordinator/group/streams/CurrentAssignmentBuilderTest.java

```diff
@@ -0,0 +1,744 @@
+        );
+
+        assertEquals(
+            member,
+            currentAssignmentBuilder
+                .withOwnedAssignment(mkTaskTuple(taskRole,
+                    mkTasks(subtopologyId1, 2, 3),
+                    mkTasks(subtopologyId2, 4, 5, 6)))
+                .build()
+        );
+    }
+
+    @ParameterizedTest
+    @EnumSource(TaskRole.class)
+    public void testUnrevokedTasksToUnrevokedTasks(TaskRole taskRole) {
+        String subtopologyId1 = Uuid.randomUuid().toString();
+        String subtopologyId2 = Uuid.randomUuid().toString();
+
+        StreamsGroupMember member = new StreamsGroupMember.Builder("member")
+            .setState(MemberState.UNREVOKED_TASKS)
+            .setProcessId("process")
+            .setMemberEpoch(10)
+            .setPreviousMemberEpoch(10)
+            .setAssignedTasks(mkTaskTuple(taskRole,
+                mkTasks(subtopologyId1, 2, 3),
+                mkTasks(subtopologyId2, 5, 6)))
+            .setTasksPendingRevocation(mkTaskTuple(taskRole,
+                mkTasks(subtopologyId1, 1),
+                mkTasks(subtopologyId2, 4)))
+            .build();
+
+        StreamsGroupMember updatedMember = new CurrentAssignmentBuilder(member)
+            .withTargetAssignment(12, mkTaskTuple(taskRole,
+                mkTasks(subtopologyId1, 3),
+                mkTasks(subtopologyId2, 6)))
+            .withCurrentActiveTaskProcessId((subtopologyId, partitionId) -> null)
+            .withOwnedAssignment(mkTaskTuple(taskRole,
+                mkTasks(subtopologyId1, 2, 3),
+                mkTasks(subtopologyId2, 5, 6)))
+            .build();
+
+        assertEquals(
+            new StreamsGroupMember.Builder("member")
+                .setState(MemberState.UNREVOKED_TASKS)
+                .setProcessId("process")
+                .setMemberEpoch(11)
+                .setPreviousMemberEpoch(10)
+                .setAssignedTasks(mkTaskTuple(taskRole,
+                    mkTasks(subtopologyId1, 3),
+                    mkTasks(subtopologyId2, 6)))
+                .setTasksPendingRevocation(mkTaskTuple(taskRole,
+                    mkTasks(subtopologyId1, 2),
+                    mkTasks(subtopologyId2, 5)))
+                .build(),
+            updatedMember
+        );
+    }
+
+    @ParameterizedTest
+    @EnumSource(TaskRole.class)
+    public void testUnrevokedTasksToUnreleasedTasks(TaskRole taskRole) {
+        String subtopologyId1 = Uuid.randomUuid().toString();
+        String subtopologyId2 = Uuid.randomUuid().toString();
+
+        StreamsGroupMember member = new StreamsGroupMember.Builder("member")
+            .setState(MemberState.UNREVOKED_TASKS)
+            .setProcessId("process")
+            .setMemberEpoch(11)
+            .setPreviousMemberEpoch(10)
+            .setAssignedTasks(mkTaskTuple(taskRole,
+                mkTasks(subtopologyId1, 2, 3),
+                mkTasks(subtopologyId2, 5, 6)))
+            .setTasksPendingRevocation(TaskTuple.EMPTY)
```

Comment:
Shouldn't this be non-empty? The member is in `UNREVOKED_TASKS`. It should have pending tasks.

Reply:
Did you forget to add this change? This line is still `.setTasksPendingRevocation(TaskTuple.EMPTY)`.

Reply:
Done